rero_ils/modules/acq_orders/serializers/csv.py

```diff
@@ -0,0 +1,283 @@
+                                            authorized_access_point]
+                                    )
+
+                    document_data['document_creator'] = ' ; '.join(creator)
+
+                    return document_data
+
+                records = DocumentsSearch() \
+                    .get_records_by_terms(terms=doc_pids)
+                return {
+                    record.pid: _build_doc(record.to_dict())
+                    for record in records
+                }
+
+            def get_accounts_by_pids(account_pids):
+                """Get accounts for the given pids."""
+                return AcqAccountsSearch()\
+                    .get_records_by_terms(terms=account_pids, as_dict=True)
+
+            def get_receipts_by_order_pids(order_pids):
+                """Get receipts for the given pids."""
+                return AcqReceiptsSearch() \
+                    .get_records_by_terms(terms=order_pids,
+                                          key='acq_order__pid',
+                                          as_dict=True)
+
+            def get_receipt_lines_by_order_line_pids(order_lines_pids):
+                """Get receipts for the given order lines pids."""
+                receipt_line_results = AcqReceiptLinesSearch() \
+                    .get_records_by_terms(terms=order_lines_pids,
+                                          key='acq_order_line__pid')
+                # organize receipt lines by oreder line pid
+                receipt_lines = {}
+                for record in receipt_line_results:
+                    key = record.acq_order_line.pid
+                    data = receipt_lines.get(key, [])
+                    data.append(record.to_dict())
+                    receipt_lines[key] = data
+                return receipt_lines
+
+            def get_vendors_by_pids(vendor_pids):
+                """Get vendors for the given pids."""
+                return VendorsSearch() \
+                    .get_records_by_terms(terms=vendor_pids, as_dict=True)
+
+            headers = dict.fromkeys(self.csv_included_fields)
+
+            # write the CSV output in memory
+            line = Line()
+            writer = csv.DictWriter(line,
+                                    quoting=csv.QUOTE_ALL,
+                                    fieldnames=headers)
+            writer.writeheader()
+            yield line.read()
+
+            for pids, order_batch_results in batch(search_result):
+                order_lines, documents, accounts, receipt_lines = \
+                    get_linked_records_by_order_pids(pids)
+                receipts = get_receipts_by_order_pids(pids)
+                # vendors
+                vendor_pids = [order['vendor']['pid']
+                               for order in order_batch_results.values()
+                               if order['vendor']['pid'] not in vendors]
+                vendors.update(get_vendors_by_pids(vendor_pids))
```

Comment:
Maybe all the vendors can be store for all batches.

Reply:
In general, it is often the same vendors who are used in orders for a library. 
Loading all vendors seems a bit overkill to me but we could do that.